crates/menmos-client/src/metadata_detector.rs

```diff
@@ -0,0 +1,68 @@
+use std::{collections::HashMap, path::Path};
+
+use serde_json;
+use snafu::{ResultExt, Snafu};
+
+use crate::Meta;
+
+#[derive(Debug, Snafu)]
+pub enum MetadataDetectorError {
+    #[snafu(display("failed to load the mime types: {}", source))]
+    MimeTypesDeserializationError { source: serde_json::Error },
+}
+
+type Result<T> = std::result::Result<T, MetadataDetectorError>;
+
+#[derive(Clone)]
+pub struct MetadataDetector {
+    mime_types: HashMap<String, String>,
+}
+
+impl MetadataDetector {
+    pub fn new() -> Result<Self> {
+        let bytes = include_bytes!("data/mime-types.json");
+
+        let mime_types = serde_json::from_slice(bytes).context(MimeTypesDeserializationError)?;
+
+        Ok(Self { mime_types })
+    }
+
+    fn detect_mime_type<P: AsRef<Path>>(&self, path: P) -> Option<&String> {
```

Comment:
Returning  a `Option<&String>` made me raise an eyebrow here.

If it were a public  method I'd be all for it, but since this is a private API I think I'd actually be in favor of returning `Option<String>` instead, because since its private we know for sure that every time this is called, the returned string will be cloned. Returning an owned string saves the `clone()` for the caller(s)